javatests/google/registry/rde/XjcToDomainResourceConverterTest.java

```diff
@@ -0,0 +1,484 @@
+import com.google.common.io.ByteSource;
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.VoidWork;
+import google.registry.model.billing.BillingEvent;
+import google.registry.model.billing.BillingEvent.Flag;
+import google.registry.model.billing.BillingEvent.Reason;
+import google.registry.model.contact.ContactResource;
+import google.registry.model.domain.DesignatedContact;
+import google.registry.model.domain.DomainResource;
+import google.registry.model.domain.GracePeriod;
+import google.registry.model.domain.rgp.GracePeriodStatus;
+import google.registry.model.domain.secdns.DelegationSignerData;
+import google.registry.model.eppcommon.StatusValue;
+import google.registry.model.host.HostResource;
+import google.registry.model.poll.PollMessage;
+import google.registry.model.reporting.HistoryEntry;
+import google.registry.testing.AppEngineRule;
+import google.registry.testing.ExceptionRule;
+import google.registry.xjc.rdedomain.XjcRdeDomain;
+import google.registry.xjc.rdedomain.XjcRdeDomainElement;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link XjcToDomainResourceConverter} */
+@RunWith(JUnit4.class)
+public class XjcToDomainResourceConverterTest {
+
+  //List of packages to initialize JAXBContext
+  private static final String JAXB_CONTEXT_PACKAGES = Joiner.on(":")
+      .join(Arrays.asList(
+          "google.registry.xjc.contact",
+          "google.registry.xjc.domain",
+          "google.registry.xjc.host",
+          "google.registry.xjc.mark",
+          "google.registry.xjc.rde",
+          "google.registry.xjc.rdecontact",
+          "google.registry.xjc.rdedomain",
+          "google.registry.xjc.rdeeppparams",
+          "google.registry.xjc.rdeheader",
+          "google.registry.xjc.rdeidn",
+          "google.registry.xjc.rdenndn",
+          "google.registry.xjc.rderegistrar",
+          "google.registry.xjc.smd"));
+
+  @Rule
+  public final AppEngineRule appEngine = AppEngineRule.builder()
+      .withDatastore()
+      .build();
+
+  @Rule
+  public final ExceptionRule thrown = new ExceptionRule();
+
+  private Unmarshaller unmarshaller;
+
+  @Before
+  public void before() throws Exception {
+    createTld("example");
+    unmarshaller = JAXBContext.newInstance(JAXB_CONTEXT_PACKAGES)
+        .createUnmarshaller();
+  }
+
+  @Test
+  public void testConvertDomainResource() throws Exception {
+    final ContactResource jd1234 = persistActiveContact("jd1234");
+    final ContactResource sh8013 = persistActiveContact("sh8013");
+    ofy().transact(new VoidWork() {
```

Comment:
Here and elsewhere, rather than wrapping everything in a transaction, wrap only the necessary parts in a transaction.  In this test, I believe only the single line that calls `XjcToDomainResourceConverter.convertDomain(xjcDomain);` needs to be in a transaction.  Everything else shouldn't be in the transaction.  Rather than using `VoidWork`, use `Work<DomainResource>`, and return the converted domain, then make assertions on it outside of the transaction.